### PR TITLE
Seguro para npcs para ciudadanos.

### PR DESCRIPTION
**Pull request by carlospaezp**

### CODE REVIEW

File: Codigo/Declares.bas [hunk 1]
```diff
@@ -1969,6 +1969,8 @@ Public Type t_UserFlags
 
     TargetUser As t_UserReference ' Usuario señalado
     
+    AttackedNpc As Integer
```
Comment (matiascalegaris) [inline Codigo/Declares.bas:1972]: dont keep reference to npc as simple integers, use t_npcref instead

File: Codigo/SistemaCombate.bas [hunk 2]
```diff
@@ -1833,6 +1833,45 @@ Public Function PuedeAtacarNPC(ByVal AttackerIndex As Integer, ByVal NpcIndex As
...
+            
+            If NpcList(NpcIndex).flags.AttackedBy <> "" Then
+                Dim AttackedBy As t_UserReference
+                Dim AttackedByIndex As Integer
+                Dim Lider As t_User
+                Dim a As Integer
+                
+
+                    AttackedBy = NameIndex(NpcList(NpcIndex).flags.AttackedBy)
+                    If IsValidUserRef(AttackedBy) Then
```
Comment (matiascalegaris) [inline Codigo/SistemaCombate.bas:1846]: this is confusing, can we replace attackedby wih CurrentOwner?

File: Codigo/Declares.bas (hunk 1 above)
Comment (carlospaezp) [reply Codigo/Declares.bas:1972]: si, lo vi, en este caso no afectaba mucho pq es simplemente para q si le pega a otro bicho tenga otro valor, pero lo puedo cambiar.

File: Codigo/SistemaCombate.bas (hunk 2 above)
Comment (carlospaezp) [reply Codigo/SistemaCombate.bas:1846]: ok

File: Codigo/SistemaCombate.bas [hunk 3]
```diff
@@ -1833,6 +1833,45 @@ Public Function PuedeAtacarNPC(ByVal AttackerIndex As Integer, ByVal NpcIndex As
...
+                        AttackedByIndex = AttackedBy.ArrayIndex
+                        If AttackedByIndex <> attackerIndex Then
+                            If UserList(AttackedByIndex).flags.AttackedNpc = NpcIndex And UserList(AttackedByIndex).flags.Muerto = 0 And Status(AttackedByIndex) = Ciudadano And (UserList(attackerIndex).GuildIndex = 0 Or UserList(attackerIndex).GuildIndex <> UserList(AttackedByIndex).GuildIndex) Then
+                                
+                                If UserList(attackerIndex).Grupo.EnGrupo Then
+                                    Lider = UserList(UserList(attackerIndex).Grupo.Lider.ArrayIndex)
```
Comment (matiascalegaris) [inline Codigo/SistemaCombate.bas:1852]: no need to loop all group members, you can test the groupo.id for both new attacker and the currentowner

File: Codigo/Declares.bas (hunk 1 above)
Comment (matiascalegaris) [reply Codigo/Declares.bas:1972]: we already keep this value at .flags.NPCAtacado, we should be using that

File: Codigo/SistemaCombate.bas [hunk 4]
```diff
@@ -1833,6 +1833,45 @@ Public Function PuedeAtacarNPC(ByVal AttackerIndex As Integer, ByVal NpcIndex As
                         Exit Function
                 End Select
             End If
+            
+            
+            If NpcList(NpcIndex).flags.AttackedBy <> "" Then
```
Comment (matiascalegaris) [inline Codigo/SistemaCombate.bas:1838]: we should move all this test into a new function and call it from Npc.s.UserCanAttackNpc

**Review by matiascalegaris — CHANGES_REQUESTED** [review]

File: Codigo/SistemaCombate.bas [hunk 5]
```diff
@@ -1273,10 +1273,10 @@ Private Sub UserDamageToUser(ByVal AtacanteIndex As Integer, ByVal VictimaIndex
                 
                     ' Mostramos en consola el golpe al atacante solo si tiene activado el chat de combate
 190                 If UserList(AtacanteIndex).ChatCombate = 1 Then
-192                     Call WriteLocaleMsg(AtacanteIndex, "210", e_FontTypeNames.FONTTYPE_INFOBOLD, .Name & "¬" & DamageStr)
+192                     Call WriteLocaleMsg(AtacanteIndex, "210", e_FontTypeNames.FONTTYPE_INFOBOLD, .name & "¬" & DamageStr)
```
Comment (matiascalegaris) [inline Codigo/SistemaCombate.bas:1276]: please remove all these case changes, it makes harder to track the real changes

File: Codigo/SistemaCombate.bas (hunk 4 above)
Comment (carlospaezp) [reply Codigo/SistemaCombate.bas:1838]: moved

**Pull request closed without merging**